src/snowflake/cli/plugins/snowpark/common.py

```diff
@@ -197,3 +205,17 @@ def format_arg(arg: Argument):
         schema=udf_sproc.schema_name,
     )
     return f"{name}({arguments})"
+
+
+def _get_import_from_resource(
+    resource_json: dict, source_name: str = "app.zip"
+) -> Set[str] | None:
+    pattern = re.compile(r"@\w+\.\w+\.([^,]+)")
+    if "imports" in resource_json.keys():
+        return {
+            imp.lower()
+            for imp in pattern.findall(resource_json["imports"])
+            if source_name not in imp
+        }
+    else:
+        return set()
```

Comment:
Unnecessary else

Reply:
WDYT about reversing the order?
```py
if "imports" not in resource_json.keys():
    return set()
return {
            imp.lower()
            for imp in pattern.findall(resource_json["imports"])
            if source_name not in imp
        }
```

Reply:
What I would recommend is building two sets:
1. Expected imports - including the sources
2. Current imports
then you can do
```py
if expected_imports.difference(current_imports):
   return True
```

If in future, this may help in optimizing uploads

Reply:
Done, however i decides to stick with `!=` comparison as it also works, when we try to remove some imports from function definition